src/bundle/Controller/LocationController.php

```diff
@@ -465,72 +464,48 @@ public function addAction(Request $request): Response
-     *
-     * @throws \InvalidArgumentException
-     * @throws TranslationInvalidArgumentException
-     * @throws APIRepositoryUnauthorizedException
-     * @throws InvalidOptionsException
+     * @return \Symfony\Component\HttpFoundation\Response
      */
     public function updateVisibilityAction(Request $request): Response
     {
         $form = $this->formFactory->updateVisibilityLocation();
         $form->handleRequest($request);
 
-        /** @var ContentInfo $contentInfo */
-        $contentInfo = $form->getData()->getLocation()->getContentInfo();
-
         if ($form->isSubmitted() && $form->isValid()) {
-            $result = $this->submitHandler->handle($form, function (LocationUpdateVisibilityData $data) use ($contentInfo) {
-                $location = $data->getLocation();
-                $hidden = $data->getHidden();
+            $data = $form->getData();
+
+            $location = $data->getLocation();
+            $hidden = $data->getHidden();
 
+            try {
                 if ($hidden) {
                     $this->locationService->hideLocation($location);
-
-                    $message = $this->translator->trans(
-                        /** @Desc("Location '%name%' hidden.") */
-                        'location.update_success.success.hidden',
-                        ['%name%' => $contentInfo->name],
-                        'location'
-                    );
                 } else {
                     $this->locationService->unhideLocation($location);
-                    $message = $this->translator->trans(
-                        /** @Desc("Location '%name%' unhidden.") */
-                        'location.update_success.success.unhidden',
-                        ['%name%' => $contentInfo->name],
-                        'location'
-                    );
                 }
-
-                $this->notificationHandler->success($message);
-
-                return new RedirectResponse($this->generateUrl('_ezpublishLocation', [
-                    'locationId' => $contentInfo->mainLocationId,
-                    '_fragment' => LocationsTab::URI_FRAGMENT,
-                ]));
-            });
-
-            if ($result instanceof Response) {
-                return $result;
+            } catch (APIRepositoryUnauthorizedException $e) {
+                return new JsonResponse(['errors' => [$e->getMessage()]], Response::HTTP_UNAUTHORIZED);
+            }
+        } else {
+            $errors = [];
+            foreach ($form->getErrors(true, true) as $formError) {
+                $errors[] = $formError->getMessage();
             }
+
+            return new JsonResponse(['errors' => $errors], Response::HTTP_UNPROCESSABLE_ENTITY);
         }
 
-        return $this->redirect($this->generateUrl('_ezpublishLocation', [
-            'locationId' => $contentInfo->mainLocationId,
-            '_fragment' => LocationsTab::URI_FRAGMENT,
-        ]));
+        return new JsonResponse();
```

Comment:
I think this should return some `success` and ideally some text for the `success` notification.